grails-app/services/org/pih/warehouse/stockTransfer/StockTransferService.groovy

```diff
@@ -167,4 +171,88 @@ class StockTransferService {
         orderItem.destinationBinLocation = stockTransferItem.destinationBinLocation
         return orderItem
     }
+
+    Order completeStockTransfer(StockTransfer stockTransfer) {
+        validateStockTransfer(stockTransfer)
+
+        // Save the stockTransfer as an order
+        Order order = createOrderFromStockTransfer(stockTransfer)
+
+        // Need to process the split items
+        processSplitItems(stockTransfer)
+
+        stockTransfer.stockTransferItems.each { StockTransferItem stockTransferItem ->
+            TransferStockCommand command = new TransferStockCommand()
+            command.location = stockTransferItem.location?:stockTransfer?.destination
+            command.binLocation = stockTransferItem.originBinLocation
+            command.inventoryItem = stockTransferItem.inventoryItem
+            command.quantity = stockTransferItem.quantity
+            command.otherLocation = stockTransferItem.location
+            command.otherBinLocation = stockTransferItem.destinationBinLocation
+            command.order = order
+            command.transferOut = Boolean.TRUE
+            command.disableRefresh = Boolean.TRUE
+            inventoryService.transferStock(command)
+        }
+
+        grailsApplication.mainContext.publishEvent(new StockTransferCompletedEvent(stockTransfer))
```

Comment:
Is this necessary? If I'm not mistaken, the transferStock method will create a transaction that will do the same thing.

Reply:
To verify this, just see whether the transfer cause the refresh to be executed twice.

Reply:
@jmiranda you are right, that was totally redundant. I removed it.